brave-grpc/src/test/java/com/github/kristofa/brave/grpc/BraveGrpcInterceptorsTest.java

```diff
@@ -0,0 +1,126 @@
+import com.twitter.zipkin.gen.BinaryAnnotation;
+import com.twitter.zipkin.gen.Span;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.ServerInterceptors;
+import io.grpc.Status;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.GreeterGrpc.GreeterBlockingStub;
+import io.grpc.examples.helloworld.GreeterGrpc.GreeterFutureStub;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class BraveGrpcInterceptorsTest {
+
+    private static final HelloRequest HELLO_REQUEST = HelloRequest.newBuilder()
+        .setName("brave")
+        .build();
+
+    private Server server;
+    private ManagedChannel channel;
+
+    @Before
+    public void before() throws Exception {
+        SpanCollectorForTesting.getInstance().getCollectedSpans().clear();
+
+        final Brave.Builder builder = new Brave.Builder();
+        Brave brave = builder
+            .spanCollector(SpanCollectorForTesting.getInstance())
+            .traceSampler(Sampler.create(1.0f))
+            .build();
+
+        int serverPort = pickUnusedPort();
+        server = ServerBuilder.forPort(serverPort)
+            .addService(ServerInterceptors.intercept(
+                GreeterGrpc.bindService(new GreeterImpl()), new BraveGrpcServerInterceptor(brave)))
+            .build()
+            .start();
+
+        channel = ManagedChannelBuilder.forAddress("localhost", serverPort)
+            .intercept(new BraveGrpcClientInterceptor("helloWorldCaller", brave))
+            .usePlaintext(true)
+            .build();
+    }
+
+    @Test
+    public void testBlockingUnaryCall() throws Exception {
+        GreeterBlockingStub stub = GreeterGrpc.newBlockingStub(channel);
+        HelloReply reply = stub.sayHello(HELLO_REQUEST);
+        assertThat(reply.getMessage(), is("Hello brave"));
+        validateSpans();
+    }
+
+    @Test
+    public void testAsyncUnaryCall() throws Exception {
+        GreeterFutureStub futureStub = GreeterGrpc.newFutureStub(channel);
+        ListenableFuture<HelloReply> helloReplyListenableFuture = futureStub.sayHello(HELLO_REQUEST);
+        HelloReply reply = helloReplyListenableFuture.get();
+        assertThat(reply.getMessage(), is("Hello brave"));
+        validateSpans();
+    }
+
+    private void validateSpans() throws Exception {
+        List<Span> spans = SpanCollectorForTesting.getInstance().getCollectedSpans();
+        assertThat(spans.size(), is(equalTo(2)));
+        assertThat(spans.get(0).getParent_id(), is(equalTo(spans.get(1).getParent_id())));
+        validateSpan(spans.get(0), Arrays.asList("ss", "sr"));
```

Comment:
maybe make a comment explaining that presence of two spans means that propagation between client and server works (ex the headers are propagated)